arkouda/dataframe.py

```diff
@@ -56,37 +58,186 @@ def groupby_operators(cls):
+        If True the grouped values of the aggregation keys will be treated as an index.
+
+    """
 
-    def __init__(self, gb, df):
+    def __init__(self, gb, df, gb_key_names=None, as_index=True):
         self.gb = gb
         self.df = df
-        for attr in ["nkeys", "size", "permutation", "unique_keys", "segments"]:
+        self.gb_key_names = gb_key_names
+        self.as_index = as_index
+
+        for attr in ["nkeys", "permutation", "unique_keys", "segments"]:
             setattr(self, attr, getattr(gb, attr))
 
     @classmethod
     def _make_aggop(cls, opname):
+        numerical_dtypes = [akfloat64, akint64, akuint64]
+
         def aggop(self, colnames=None):
-            if isinstance(colnames, str):
-                return Series(self.gb.aggregate(self.df.data[colnames], opname))
-            else:
-                if colnames is None:
-                    colnames = list(self.df.data.keys())
-                if isinstance(colnames, List):
+            """
+            Aggregate the operation, with the grouped column(s) values as keys.
+
+            Parameters
+            ----------
+
+            colnames : (list of) str, default=None
+                Column name or list of column names to compute the aggregation over.
+
+            Returns
+            -------
+            arkouda.dataframe.DataFrame
+
+            """
+            if colnames is None:
+                colnames = list(self.df.data.keys())
+            elif isinstance(colnames, str):
+                colnames = [colnames]
+            colnames = [
+                c
+                for c in colnames
+                if (
+                    (self.df.data[c].dtype.type in numerical_dtypes)
+                    or isinstance(self.df.data[c].dtype, BigInt)
+                )
+                and (
+                    (isinstance(self.gb_key_names, str) and (c != self.gb_key_names))
+                    or (isinstance(self.gb_key_names, list) and c not in self.gb_key_names)
+                )
+            ]
+
+            if isinstance(colnames, List):
+                if isinstance(self.gb_key_names, str):
                     return DataFrame(
                         {c: self.gb.aggregate(self.df.data[c], opname)[1] for c in colnames},
-                        index=self.gb.unique_keys,
+                        index=Index(self.gb.unique_keys, name=self.gb_key_names),
                     )
+                elif isinstance(self.gb_key_names, list) and len(self.gb_key_names) == 1:
+                    return DataFrame(
+                        {c: self.gb.aggregate(self.df.data[c], opname)[1] for c in colnames},
+                        index=Index(self.gb.unique_keys, name=self.gb_key_names[0]),
+                    )
+                elif isinstance(self.gb_key_names, list):
+                    column_dict = {
+                        self.gb_key_names[i]: self.unique_keys[i] for i in range(len(self.gb_key_names))
+                    }
```

Comment:
purely stylistic suggestion, so no need to change if u dont want to. I think you can accomplish the same thing with
```python
column_dict = dict(zip(self.gb_key_names, self.unique_keys))
```
which is a bit more pythonic

Reply:
This is better.  Thanks for the tip.  I made the change.